image-patching/stopgap-script/nuage_overcloud_full_patch.py

```diff
@@ -151,239 +168,296 @@ def rhel_subscription(username, password, pool, image,
-                              args.ImageName, args.ProxyHostname,
-                              args.ProxyPort)
+        if nuage_config.get("ProxyHostname") and nuage_config.get("ProxyPort"):
+            rhel_subscription(
+                nuage_config["RhelUserName"], nuage_config["RhelPassword"],
+                nuage_config["RhelPool"], nuage_config["ImageName"],
+                nuage_config["ProxyHostname"], nuage_config["ProxyPort"])
         else:
-            rhel_subscription(args.RhelUserName, args.RhelPassword,
-                              args.RhelPool,
-                              args.ImageName)
-
+            rhel_subscription(
+                nuage_config["RhelUserName"], nuage_config["RhelPassword"],
+                nuage_config["RhelPool"], nuage_config["ImageName"])
     cmds_run(['echo "Uninstalling packages"'])
-    uninstall_packages(args.ImageName)
+    uninstall_packages(nuage_config["ImageName"])
+
+    cmds_run(['echo "Copying RepoFile to the overcloud image"'])
+    copy_repo_file(nuage_config["ImageName"], nuage_config["RepoFile"])
 
-    cmds_run(['echo "Creating Repo File"'])
-    create_repo_file(args.RepoName, args.RepoBaseUrl, args.ImageName,
-                     GPGCHECK)
+    if "ovrs" in nuage_config["DeploymentType"]:
+        cmds_run(['echo "Updating Kernel Packages"'])
+        update_kernel(nuage_config["ImageName"],
+                      nuage_config["KernelRepoName"])
+        cmds_run(['echo "Updating MLNX OFED Packages"'])
+        install_mellanox(nuage_config["ImageName"], workingDir,
+                         nuage_config["MellanoxRepoName"])
 
-    cmds_run(['echo "Updating Kernel"'])
-    update_kernel(args.ImageName)
+    if "avrs" in nuage_config["DeploymentType"]:
+        cmds_run(['echo "Downloading AVRS Packages"'])
+        copy_avrs_packages(nuage_config["ImageName"],
+                           nuage_config["AVRSRepoName"])
 
     cmds_run(['echo "Installing Nuage Packages"'])
-    install_packages(args.ImageName)
+    install_nuage_packages(nuage_config["ImageName"],
+                           nuage_config["VRSRepoName"])
 
-    cmds_run(['echo "Installing MLNX OFED Packages"'])
-    install_mellanox(args.ImageName, workingDir)
+    if nuage_config.get("RhelUserName") and nuage_config.get(
+            "RhelPassword") and nuage_config.get("RhelPool"):
+        rhel_remove_subscription(nuage_config["ImageName"])
 
-    cmds_run(['echo "Cleaning up"'])
-    delete_repo_file(args.ImageName)
+    cmds_run(['echo "Done"'])
 
-    if args.RhelUserName and args.RhelPassword and args.RhelPool:
-        rhel_remove_subscription(args.ImageName)
 
-    cmds_run(['echo "Done"'])
+def check_config(nuage_config):
+    for key in ["ImageName", "DeploymentType", "RepoFile", "VRSRepoNames"]:
+        if not (nuage_config.get(key)):
+            logger.error("Please provide missing config %s value "
+                         "in your config file. \n" % key)
```

Comment:
can we change the logic in such a way that the code will check all these params and tells which all are missed. Instead of erroring out right after 1 param is not found.